### ISSUE TITLE
Weird behavior with bridging

### ISSUE THREAD

**Issue by thomaspj1017** [supermerill/SuperSlicer#1398]
### Version
SuperSlicer Version 2.3.56.6

### Operating system type + version
Windows 10

### Behavior
For parts with a one layer thick wall beneath an unsupported hole, rather than do straight bridging across, it creates a bridge circle which defeats the purpose of the sacrificial layer.

Below is a screen shot with the one on the left showing the issue and the one on the right showing how I want it/it should perform for unsupported vertical holes.

![image](https://user-images.githubusercontent.com/16673263/124821621-8f46b000-df34-11eb-9a68-bd33066ca5db.png)

**Comment by supermerill** [comment]
Untick "only one perimeter on top" or at least at this layer, with a 'height range modifier'

Because it's a one-layer surface, it's tagged as "top", and so the algorithm shift other perimeters.
You're the second one to report that.
Maybe I can to check for bridging before adding these perimeters.

**Comment by Zeblote** [comment]
I just encountered this strange behavior aswell.

Prusa Slicer:
![image](https://user-images.githubusercontent.com/6764788/147287752-553d3578-6267-4d43-8f3b-9aa20e6e0be4.png)

Super Slicer:
![image](https://user-images.githubusercontent.com/6764788/147287732-d64c5cf2-04ee-4bb8-80fa-611b5163f80e.png)

Unchecking "only one perimeter" "on top surfaces" fixed it. However, this is not a top surface!
![image](https://user-images.githubusercontent.com/6764788/147287803-b512f0fb-4ded-49e7-a918-ccd525a4ae03.png)

**Comment by supermerill** [comment]
@Zeblote
you forgot to add your project, so i can't test, but I guess that the difference here is from the 'anchor solid infill'  bridge margin that is set to 3mm in prusaslicer (and also the default one).

**Comment by Zeblote** [comment]
Good point. The file I was trying to slice is here: https://main.eva-3d.page/hotends/rapido/

And the project: [bridge_infill_bug.zip](https://github.com/supermerill/SuperSlicer/files/7771994/bridge_infill_bug.zip)

**Comment by supermerill** [comment]
> However, this is not a top surface!

Not a top surface but there's nothing on top of it. 
![image](https://user-images.githubusercontent.com/6536403/151780318-c3f24770-2f5d-4d15-aeeb-57a7f7b15219.png)

**Comment by supermerill** [comment]
for the "overhang" detection, it's deactivated in the linked project. You just ahve to reactivate it.
for the problematic bridging area, it's a duplicate of issue #1494

**Comment by OldCurmudgeon3DP** [comment]
I've been running into the same issue as the OP when trying to avoid using supports for vertical holes in parts. I'll try unchecking the 1 perimeter option. I found that it does it using both the built-in primitives and if I have a 1 layer thick "bridge" built in from the modeling software (Fusion).
